testkit/src/integration-testkit.ts

```diff
@@ -45,32 +49,60 @@ export namespace IntegrationTestkit {
   export type ShutdownCallback = (error?: Error) => void;
 }
 
+/**
+ * @public
+ */
+export interface IntegrationTestKitOptions extends KalixOptions {
+  /**
+   * The name of this service when deployed
+   */
+  nameOfService?: String;
+  /**
+   * Whether ACL checking is enabled.
+   */
+  aclCheckingEnabled?: boolean;
+  /**
+   * Docker image to test
+   */
+  dockerImage?: string;
+}
+
 /**
  * Integration Testkit.
  *
  * @public
  */
 export class IntegrationTestkit {
-  private options: any = { dockerImage: defaultDockerImage };
+  private options: IntegrationTestKitOptions;
   private componentClients: { [serviceName: string]: any };
   private kalix: Kalix;
   private proxyContainer: any;
+  private proxyPort?: number;
 
   /**
    * Integration Testkit.
    *
    * @param options - Options for the testkit and Kalix service
    */
-  constructor(options?: any) {
+  constructor(options?: IntegrationTestKitOptions) {
     if (options) {
       this.options = options;
-      if (!options.dockerImage) {
-        this.options.dockerImage = defaultDockerImage;
-      }
+    } else {
+      this.options = {};
+    }
+
+    this.options.delayDiscoveryUntilProxyPortSet = true;
+
+    // sensible defaults for missing values
+    if (!this.options.dockerImage) {
+      this.options.dockerImage = defaultDockerImage;
+    }
+    if (!this.options.serviceName) {
+      this.options.serviceName = 'self';
```

Comment:
There's `nameOfService` in `IntegrationTestKitOptions` and then `options.serviceName` here. Are they meant to be the same option?

Reply:
Ah, indeed, we should only have `serviceName`.